### PR TITLE
Added mention of double parsing and extended reject reason

### PR DESCRIPTION
**Pull request by martinheidegger**
Following the suggestions of @mikesherov in #12 this PR adds a better explanation of double parsing to the content-sniffing.

### CODE REVIEW

**Comment by mikesherov** [comment]
👍

File: readme.md [hunk 1]
```diff
@@ -104,12 +104,13 @@ special file ending. Several file endings have been discussed, in the end
...
 
-### 3) ~~Content-Sniffing~~ _(rejected)_
+### 3) ~~Content-Sniffing aka. Double Parsing~~ _(rejected)_
 
 The content would be pre-parsed to identify whether its a `CommonJS` or
 `ES2015` module.
 
-_(Rejected because the detection is fuzzy and could lead to user confusion and problems with tooling)_
+_(Rejected because the detection has ambiguity in `CommonJS` packages. This could be revisited if the TC39 changes the specification of ES2015 modules but
+could still be rejected because of performance reasons)_
```
Comment (ljharb) [inline readme.md]: it could also still be rejected because not every environment that needs to disambiguate module vs script has a JS parser available.

Comment (mikesherov) [inline readme.md]: @ljharb I attempted to address that in the other thread. Not really double-parse, double-execute.

Comment (ljharb) [inline readme.md]: Sure - but I still think that this note should mention all the reasons it is still likely to be rejected, even in the event of spec changes.

**Comment by martinheidegger** [comment]
Your discussion made rethink the phrasing and it is now written in a way that whoever is interested in the exact reason will click on the `[more here]` link. If you have problems with the wording, please provide a PR.

**Pull request merged**